docs/source/cpp/compute.rst

```diff
@@ -1816,25 +1816,25 @@ Structural transforms
   The output type is an Array of items for the ``FIRST``/``LAST`` options
   and an Array of List of items for the ``ALL`` option.
 
-* \(6) Extract a child value based on a sequence of indices passed in
+* \(6) Extract a child value based on a field name or a sequence of indices passed in
   the options. The validity bitmap of the result will be the
   intersection of all intermediate validity bitmaps. For example, for
   an array with type ``struct<a: int32, b: struct<c: int64, d:
   float64>>``:
 
-  * An empty sequence of indices yields the original value unchanged.
-  * The index ``0`` yields an array of type ``int32`` whose validity
+  * An empty sequence of indices or an empty field name yields the original value unchanged.
```

Comment:
Are we sure an "empty field name" does that? (it could also give an error because the name "" (empty string) is not present in the schema)